plugins/inputs/mysql/mysql.go

```diff
@@ -215,25 +657,677 @@ func (m *Mysql) gatherServer(serv string, acc telegraf.Accumulator) error {
+	rows, err := db.Query(perfFileEventsQuery)
+	if err != nil {
+		return err
+	}
+
+	defer rows.Close()
+
+	var (
+		eventName                                 string
+		countRead, countWrite, countMisc          uint64
+		sumTimerRead, sumTimerWrite, sumTimerMisc uint64
+		sumNumBytesRead, sumNumBytesWrite         uint64
+	)
+
+	servtag, err := parseDSN(serv)
+	if err != nil {
+		servtag = "localhost"
+	}
+	tags := map[string]string{
+		"server": servtag,
+	}
+	for rows.Next() {
+		err = rows.Scan(
+			&eventName,
+			&countRead, &sumTimerRead, &sumNumBytesRead,
+			&countWrite, &sumTimerWrite, &sumNumBytesWrite,
+			&countMisc, &sumTimerMisc,
+		)
+		if err != nil {
+			return err
+		}
+
+		tags["event_name"] = eventName
+		fields := make(map[string]interface{})
+
+		tags["mode"] = "misc"
+		fields["file_events_total"] = float64(countWrite)
+		fields["file_events_seconds_total"] = float64(sumTimerMisc) / picoSeconds
+		acc.AddFields("mysql_perf_schema", fields, tags)
+
+		tags["mode"] = "read"
+		fields["file_events_total"] = float64(countRead)
+		fields["file_events_seconds_total"] = float64(sumTimerRead) / picoSeconds
+		fields["file_events_bytes_totals"] = float64(sumNumBytesRead)
+		acc.AddFields("mysql_perf_schema", fields, tags)
+
+		tags["mode"] = "write"
+		fields["file_events_total"] = float64(countWrite)
+		fields["file_events_seconds_total"] = float64(sumTimerWrite) / picoSeconds
+		fields["file_events_bytes_totals"] = float64(sumNumBytesWrite)
+		acc.AddFields("mysql_perf_schema", fields, tags)
+
+	}
+	return nil
+}
+
+// gatherPerfEventsStatements can be used to get attributes of each event
+func (m *Mysql) gatherPerfEventsStatements(db *sql.DB, serv string, acc telegraf.Accumulator) error {
+	query := fmt.Sprintf(
+		perfEventsStatementsQuery,
+		m.PerfEventsStatementsDigestTextLimit,
+		m.PerfEventsStatementsTimeLimit,
+		m.PerfEventsStatementsLimit,
+	)
+
+	rows, err := db.Query(query)
+	if err != nil {
+		return err
+	}
+
+	defer rows.Close()
+
+	var (
+		schemaName, digest, digest_text      string
+		count, queryTime, errors, warnings   uint64
+		rowsAffected, rowsSent, rowsExamined uint64
+		tmpTables, tmpDiskTables             uint64
+		sortMergePasses, sortRows            uint64
+		noIndexUsed                          uint64
+	)
+
+	servtag, err := parseDSN(serv)
+	if err != nil {
+		servtag = "localhost"
+	}
+	tags := map[string]string{
+		"server": servtag,
+	}
+
+	for rows.Next() {
+		err = rows.Scan(
+			&schemaName, &digest, &digest_text,
+			&count, &queryTime, &errors, &warnings,
+			&rowsAffected, &rowsSent, &rowsExamined,
+			&tmpTables, &tmpDiskTables,
+			&sortMergePasses, &sortRows,
+		)
+
+		if err != nil {
+			return err
+		}
+		tags["schema"] = schemaName
+		tags["digest"] = digest
+		tags["digest_text"] = digest_text
+
+		fields := make(map[string]interface{})
+
+		fields["events_statements_total"] = float64(count)
```

Comment:
please change all instances of this, to this:

```go
fields := map[string]interface{}{
   "events_statements_seconds_total": float64(queryTime) / picoSeconds
  ...
}
```

this will create less allocations